src/v/cluster/log_eviction_stm.cc

```diff
@@ -189,6 +189,17 @@ log_eviction_stm::do_write_raft_snapshot(model::offset truncation_point) {
       truncation_point);
     auto snapshot_data = co_await _raft->stm_manager()->take_snapshot(
       truncation_point);
+    const auto start_offset = _raft->start_offset();
+    if (start_offset >= truncation_point) {
```

Comment:
This doesn't look right - `truncation_point` corresponds to a snapshot offset which is `start_offset - 1`. Should it be `>` instead?